ledger/internal/applications.go

```diff
@@ -226,13 +226,13 @@ func (cs *roundCowState) NewBox(appIdx basics.AppIndex, key string, size uint64)
 		return fmt.Errorf("box %s exists for %d", key, appIdx)
 	}
 
-	record, err := cs.Get(appIdx.Address(), false)
+	record, err := cs.Get(appAddr, false)
```

Comment:
one dumb question, why we can previously infer application address from appidx, but now we have to pass in `appAddr`? Does the previous implementation need to be corrected?

Reply:
This is an optimization.  `appIdx.Address()` computes a hash.  There is a cache of such things on the calling side, so we pass it in instead.  The caller _must_ supply the address that corresponds to this app, or the MBR will get charged to the wrong account.

Reply:
I see, so what we know is that caller maintains a cache containing such address, so we do not need to re-compute.

Another thing, what is MBR standing for?